secondCommit/src/command/CommandManager.java

```diff
@@ -0,0 +1,74 @@
+package command;
+
+import shop.IShop;
+import user.IInventoryProvider;
+
+import java.util.ArrayList;
+
+/**
+ * Command manager class is singleton class and manages  commands and shops
+ * and remember the executed commands history
+ */
+public class CommandManager {
+    //instance is private to ban the creation more than one objects
+    private static CommandManager instance;
+    //history array is for remembering  the executed commands
+    ArrayList<ICommand> history = new ArrayList<>();
+    //cursor for indicating current position in the history array
+    private int cursor = 0;

```

Comment:
historyCursor would be a better name. Cursor can be applied to many things, and does not tell us that its being used for history position. Although you have documented it, we would have to go back to the top of the file to read what this cursor is for.